src/Language/PureScript/Ide/Filter.hs

```diff
@@ -77,6 +79,15 @@ declarationFilter :: (IdeDeclaration -> Text -> Bool) -> Text -> [Module] -> [Mo
 declarationFilter predicate search =
   filterModuleDecls (flip predicate search)
 
+-- | Only keeps Identifiers in the given type declarations
+declarationTypeFilter :: [D.IdeDeclaration] -> Filter
+declarationTypeFilter [] = mkFilter identity
+declarationTypeFilter decls =
+    mkFilter $ filterModuleDecls filterDecls
+    where
+      filterDecls :: IdeDeclaration -> Bool
+      filterDecls decl = D.typeDeclarationForDeclaration decl `elem` decls
```

Comment:
Oh yes sorry, the type would be `[Module] -> [Module]`. The idea for creating the new module was so that we could split out the logic around the new declaration filters, because I intend to continue adding to them in the new future. If think it's totally fine to move `filterModuleDecls` into `Ide.Filter.Declaration` and import it from `Ide.Filter`.